examples/terminal.rs

```diff
@@ -134,6 +134,8 @@ pub fn exit() {
     terminal.exit();
 }
 
-fn main() {
-    let _ = scroll_down();
+fn main() -> io::Result<()> {
+    scroll_down()?;
```

Comment:
You can remove `?;` and `Ok(())`. The only line here will be `scroll_down()`.

Reply:
Of course I can, but should I? I think eliding the `?; Ok(())` is less clear, especially in functions that do multiple things where the last thing is mysteriously missing a `?` just because it’s last. It also prevents error conversion, e.g. from `io::Result` to `crossterm::Result`.

If you still disagree, I’ll change it (or you can).

Reply:
I am neutral in this :)

Reply:
My thoughts ...

* What is the reason to return `io::Result` instead of `crossterm::Result`?

... nitpicks ...

* Why to add another import `use std::io`?
* There's nothing mysterious about how Rust works in terms of the last expression and a function result
* The example is 20 lines shorter

Reply:
P.S. I do not want to argue much about this, if Timon is okay with this, let be it. I'm just trying to make it shorter, especially in such a simple case where there's a function like `set_terminal_size` which calls just `set_size`. I'm also aware that you didn't introduce `io::Result`, ... in this PR.

Reply:
> * What is the reason to return `io::Result` instead of `crossterm::Result`?

`io::Result` is more specific than `crossterm::Result`; if that’s the only kind of error that can happen, we shouldn’t force callers to handle the other unused cases.

Reply:
> if that’s the only kind of error that can happen

As user of the `crossterm` crate, you don't know this, do you?

Look at this example (`examples/terminal.rs`).

```rust
pub fn set_terminal_size() -> io::Result<()> {
    let terminal = terminal();
    terminal.set_size(10, 10)
}
```

* As a user, I see that the `terminal.set_size` returns `Result<(), ErrorKind>`
* What if the `set_size` will return `ErrorKind::ResizingTerminalFailure`?
* How can I, as a user, know that it wont return `ResizingTerminalFailure`?
* And if it will, you'll end up with the _can not convert error to IO error_ error, which is a bit unfriendly

```rust
impl From<ErrorKind> for io::Error {
    fn from(e: ErrorKind) -> io::Error {
        match e {
            ErrorKind::IoError(io) => return io,
            _ => io::Error::new(io::ErrorKind::Other, "can not convert error to IO error"),
        }
    }
}
```

Reply:
Let's not pollute this PR more. It can be merged and discussed elsewhere (Discord, GH issue).

Reply:
That is indeed a mistake. (Perhaps `crossterm::Error` should not be convertable to `io::Error`?)

Reply:
As this particular issue (and main reason for my comments) is out of scope of this PR (you didn't introduce this), I'm going to create an issue for this. Sounds okay?

Reply:
Trying to remove that conversion `impl` reveals that the existing code is not consistent, so I’m not sure what the right thing is. I’ve updated my PR to use `crossterm::Result`, except in the one example that already uses `io::Result` everywhere.

Reply:
I made [a generic issue](https://github.com/TimonPost/crossterm/issues/229) for this problem as it's unclear to me why the conversions exists, why we're using it in examples, etc.